packages/optimizer/lib/transformers/SeparateKeyframes.js

```diff
@@ -60,8 +68,19 @@ class SeparateKeyframes {
       },
     };
 
+    const isInvalidKeyframe = (keyframe) => {
+      return keyframe.keyframes.some((k) => {
+        return k.declarations.some((d) => {
+          return !allowedKeyframeProps[d.property];
+        });
+      });
+    };
+
     cssTree.stylesheet.rules = cssTree.stylesheet.rules.filter((rule) => {
       if (rule.type === 'keyframes') {
+        // We can't move a keyframe with an invalid property
+        // or else the style[amp-keyframes] is invalid
+        if (isInvalidKeyframe(rule)) return true;
```

Comment:
Can you log a warning here:

```
const log = require('../log').tag('SeparateKeyframes');
...

log.warn('...').
```

Ideally, with some context making it easy to identify the invalid keyframe.

Reply:
Updated